components/html-block/html-block.js

```diff
@@ -250,8 +271,22 @@ class HtmlBlock extends RtlMixin(LitElement) {
 	}
 
 	async _handleSlotChange(e) {
-		if (!e.target) return;
-		await this._render(e.target);
+		if (!e.target || !this.shadowRoot) return;
+		const slot = this.shadowRoot.querySelector('slot');
+		const slottedNodes = slot.assignedNodes({ flatten: true });
+
+		if (this._hasSlottedElements(slottedNodes)) {
+			this._hasSlottedContent = true;
+			await this.updateComplete;
+			await this._render(e.target);
+		} else {
+			this._hasSlottedContent = false;
+		}
+	}
+
+	_hasSlottedElements(slottedNodes) {
```

Comment:
Thinking about the element requirement here: the whole purpose of them providing slotted content is to avoid deferred rendering, in which case they _would_ always be providing an element. i.e. if they are just providing text, there's no reason for them not to use the `html` property.

Reply:
Right, this check exists because Lit seems to jam an empty text node into the slot. I figured it would be safe enough for the reasons you mentioned... If there's no actual HTML in here, I don't think we care about it because it's not really a valid use case for the slot (at least moving forward).